stan/math/rev/arr/functor/coupled_ode_system.hpp

```diff
@@ -338,7 +336,7 @@ struct coupled_ode_system<F, var, double> {
   std::vector<double> initial_state() const {
     std::vector<double> initial(size_, 0.0);
     for (size_t i = 0; i < N_; i++)
-      initial[i] = y0_dbl_[i];
+      initial[i] = value_of(y0_[i]);
```

Comment:
This makes sense. This is only used once, so no need to stash the initial conditions as a separate vector.